### ISSUE TITLE
Iwslt-envi-luong config fails to learn when using GRU instead of LSTM

### ISSUE THREAD

**Issue by voslucas** [joeynmt/joeynmt#61]
Amazing tool! Really usefull for fast testing / prototyping. I use it to test different RNN cell's. Thanx!

I played with both the _iwslt-envi-luong_ and the _iwslt-envi-xnmt_ predefined configs and get the results as reported. But, when i change the rnn_type in both encoder/decoder in both configs from LSTM to GRU,  a strange things happen. The _iwslt-envi-xnmt_ still learns well, no problem.. But the iwslt-envi-luong fails to learn at all.

I've included the validations.txt files for both LSTM and GRU 

What am i missing? Is there any explanation for this behaviour? Is it due to the luong attention mechanism/internals/implementation only working with LSTM? 

[validations-gru.txt](https://github.com/joeynmt/joeynmt/files/3439524/validations-gru.txt)
[validations-lstm.txt](https://github.com/joeynmt/joeynmt/files/3439525/validations-lstm.txt)

**Comment by bastings** [comment]
Hi, and thanks!

This might have to do with the optimizer and learning rate. Luong's configuration was specifically tuned for the LSTM, with SGD and lr=1.0. 
This is not a setup that is used very often. Could you try that configuration using Adam as optimizer and a learning rate of, say, 0.0003?

**Comment by voslucas** [comment]
Thank you! I will give it a shot. It is running with lr=0.003 and adam now. I keep you posted.

Yes, it could be something with the dynamics. Although with the _iwslt-envi-xnmt_ setup, the gru is comparable to the lstm out-of-the-box.

**Comment by voslucas** [comment]
Yes, these setting work. Thank you!
(Correction on previous post, i meant lr=0.0003 and not 0.003)

I'm doing some research on lstm/gru and other variants. Although the GRU sometimes stay behind and sometimes wins, i never saw a GRU really fail when given the same network/train settings as LSTM..
